src/ae/contract.ts

```diff
@@ -0,0 +1,212 @@
+    contractId,
+    queryId
+  }: {
+    contractId: string
+    queryId?: string
+  }, opt: { onAccount?: string | object } = {}): Promise<string> {
+    return await this.delegateSignatureCommon([queryId ?? await this.address(opt), contractId], opt)
+  }
+
+  /**
+   * Get network Id
+   * @instance
+   * @function getNetworkId
+   * @category async
+   * @rtype () => networkId: String
+   * @returns Network Id
+   */
+  readonly getNetworkId = getNetworkId
+
+  /**
+   * Generate contract ACI object with predefined js methods for contract usage - can be used for
+   * creating a reference to already deployed contracts
+   * @alias module:@aeternity/aepp-sdk/es/contract/aci
+   * @param options Options object
+   * @returns JS Contract API
+   * @example
+   * const contractIns = await aeSdk.getContractInstance({ source })
+   * await contractIns.deploy([321]) or await contractIns.methods.init(321)
+   * const callResult = await contractIns.call('setState', [123]) or
+   * await contractIns.methods.setState.send(123, options)
+   * const staticCallResult = await contractIns.call('setState', [123], { callStatic: true }) or
+   * await contractIns.methods.setState.get(123, options)
+   * Also you can call contract like: await contractIns.methods.setState(123, options)
+   * Then sdk decide to make on-chain or static call(dry-run API) transaction based on function is
+   * stateful or not
+   */
+  readonly getContractInstance = getContractInstance
+
+  /**
+   * Sign data blob
+   * @instance
+   * @abstract
+   * @category async
+   * @param data - Data blob to sign
+   * @param options
+   * @returns Signed data blob
+   */
+  abstract sign (data: string | Buffer, options?: object): Promise<Uint8Array>
+
+  /**
+      * Obtain account address
+      * @instance
+      * @abstract
+      * @category async
+      * @return {String} Public account address
+      */
+  abstract address (opt?: object): Promise<string>
+}
+
+/**
+ * Contract Stamp
+ *
+ * Provide contract implementation
+ * {@link module:@aeternity/aepp-sdk/es/ae--Ae} clients.
+ * @alias module:@aeternity/aepp-sdk/es/ae/contract
+ * @returns Contract instance
+ * @example
+ * import { Transaction, MemoryAccount, ChainNode } from '@aeternity/aepp-sdk
+ *
+ * const ContractWithAE = await Contract
+ *    .compose(Transaction, MemoryAccount, ChainNode) // AE implementation
+ * const client = await ContractWithAe({ url, compilerUrl, keypair, ... }) *
+ */
+
+export default Ae.compose<_Contract>(ContractCompilerHttp, {
+  methods: {
+    getContractInstance,
+    // Delegation for contract
+    delegateSignatureCommon: _Contract.prototype.delegateSignatureCommon,
+    // AENS
+    createAensDelegationSignature: _Contract.prototype.createAensDelegationSignature,
+    // Oracle
+    createOracleDelegationSignature: _Contract.prototype.createOracleDelegationSignature
+  },
+  deepProps: {
+    Ae: {
+      defaults: {
+        gasPrice: MIN_GAS_PRICE,
+        amount: AMOUNT
```

Comment:
try to inline these defaults into places where they actually needed, `AMOUNT` probably can be removed at all because it is 0